src/components/common/FiatValue/index.tsx

```diff
@@ -18,10 +26,26 @@ const FiatValue = ({ value, maxLength }: { value: string | number; maxLength?: n
     return formatCurrencyPrecise(value, currency)
   }, [value, currency])
 
+  const [whole, decimals] = useMemo(() => {
+    const match = preciseFiat.match(/(.+)(\D\d+)$/)
```

Comment:
I think we have to consider that this separator can be different across languages, either comma or period.
<img width="786" alt="Screenshot 2024-08-05 at 12 15 57" src="https://github.com/user-attachments/assets/8cd43062-c322-40e5-87d0-92b29f9a51e0">

Reply:
That's why it matches `\D`.

Reply:
You are right. It didn't work in my example because of the € at the end but in our code the `preciseFiat` doesn't contain the currency symbol.

Reply:
I am actually running into this issue because for me the $ sign is at the end of the value and not in front.

<img width="180" alt="Screenshot 2024-08-05 at 12 44 57" src="https://github.com/user-attachments/assets/8d490a4c-3d95-4b1f-9a33-20bad4cd3092">

Reply:
Fixed, thanks!